internal/controller/health/registry.go

```diff
@@ -0,0 +1,23 @@
+package health
+
+// RegisterChecker adds a Checker to the package's internal registry.
+func RegisterChecker(checker Checker) {
+	checkerReg.register(checker)
+}
+
+// checkerReg is a registry of Checkers.
+var checkerReg = checkerRegistry{}
+
+// checkerRegistry is a registry of Checkers.
+type checkerRegistry map[string]Checker
+
+// register adds a Checker to the checkerRegistry.
+func (c checkerRegistry) register(checker Checker) {
+	c[checker.Name()] = checker
+}
+
+// getChecker returns the Checker for the health check with the given name. If
+// no Checker is registered with the given name, nil is returned instead.
+func (c checkerRegistry) getChecker(name string) Checker {
+	return c[name]
+}
```

Comment:
Order of things is a bit odd, probably easier to read like this:

```suggestion
package health

// checkerRegistry is a registry of Checkers.
type checkerRegistry map[string]Checker

// checkerReg is a registry of Checkers.
var checkerReg = checkerRegistry{}

// register adds a Checker to the checkerRegistry.
func (c checkerRegistry) register(checker Checker) {
	c[checker.Name()] = checker
}

// getChecker returns the Checker for the health check with the given name. If
// no Checker is registered with the given name, nil is returned instead.
func (c checkerRegistry) getChecker(name string) Checker {
	return c[name]
}

// RegisterChecker adds a Checker to the package's internal registry.
func RegisterChecker(checker Checker) {
	checkerReg.register(checker)
}
```

Reply:
Agree it was odd. I would prefer this very slight alteration to your own suggestion as it keeps `checkerRegistry`'s methods defined immediately after the type def instead of putting the variable declaration in between.

```suggestion
package health

// checkerRegistry is a registry of Checkers.
type checkerRegistry map[string]Checker

// register adds a Checker to the checkerRegistry.
func (c checkerRegistry) register(checker Checker) {
	c[checker.Name()] = checker
}

// getChecker returns the Checker for the health check with the given name. If
// no Checker is registered with the given name, nil is returned instead.
func (c checkerRegistry) getChecker(name string) Checker {
	return c[name]
}

// checkerReg is a registry of Checkers.
var checkerReg = checkerRegistry{}

// RegisterChecker adds a Checker to the package's internal registry.
func RegisterChecker(checker Checker) {
	checkerReg.register(checker)
}
```